agent/dns_test.go

```diff
@@ -3592,86 +3589,93 @@ func TestDNS_ConfigReload(t *testing.T) {
-
-			for _, s := range a.dnsServers {
-				cfg := s.config.Load().(*dnsConfig)
-				require.Equal(t, []string{"8.8.8.8:53"}, cfg.Recursors)
-				require.Equal(t, structs.RecursorStrategy("sequential"), cfg.RecursorStrategy)
-				require.False(t, cfg.AllowStale)
-				require.Equal(t, 20*time.Second, cfg.MaxStale)
-				require.Equal(t, 10*time.Second, cfg.NodeTTL)
-				ttl, _ := cfg.GetTTLForService("my_services_1")
-				require.Equal(t, 5*time.Second, ttl)
-				ttl, _ = cfg.GetTTLForService("my_specific_service")
-				require.Equal(t, 30*time.Second, ttl)
-				require.False(t, cfg.EnableTruncate)
-				require.False(t, cfg.OnlyPassing)
-				require.Equal(t, 15*time.Second, cfg.RecursorTimeout)
-				require.False(t, cfg.DisableCompression)
-				require.Equal(t, 1, cfg.ARecordLimit)
-				require.False(t, cfg.NodeMetaTXT)
-				require.Equal(t, uint32(1), cfg.SOAConfig.Refresh)
-				require.Equal(t, uint32(2), cfg.SOAConfig.Retry)
-				require.Equal(t, uint32(3), cfg.SOAConfig.Expire)
-				require.Equal(t, uint32(4), cfg.SOAConfig.Minttl)
-			}
-
-			newCfg := *a.Config
-			newCfg.DNSRecursors = []string{"1.1.1.1:53"}
-			newCfg.DNSAllowStale = true
-			newCfg.DNSMaxStale = 21 * time.Second
-			newCfg.DNSNodeTTL = 11 * time.Second
-			newCfg.DNSServiceTTL = map[string]time.Duration{
-				"2_my_services*":        6 * time.Second,
-				"2_my_specific_service": 31 * time.Second,
-			}
-			newCfg.DNSEnableTruncate = true
-			newCfg.DNSOnlyPassing = true
-			newCfg.DNSRecursorStrategy = "random"
-			newCfg.DNSRecursorTimeout = 16 * time.Second
-			newCfg.DNSDisableCompression = true
-			newCfg.DNSARecordLimit = 2
-			newCfg.DNSNodeMetaTXT = true
-			newCfg.DNSSOA.Refresh = 10
-			newCfg.DNSSOA.Retry = 20
-			newCfg.DNSSOA.Expire = 30
-			newCfg.DNSSOA.Minttl = 40
-
-			err := a.reloadConfigInternal(&newCfg)
-			require.NoError(t, err)
+	`)
+	defer a.Shutdown()
+	testrpc.WaitForLeader(t, a.RPC, "dc1")
+
+	for _, s := range a.dnsServers {
+		server, ok := s.(*DNSServer)
+		require.True(t, ok)
+
+		cfg := server.config.Load().(*dnsConfig)
+		require.Equal(t, []string{"8.8.8.8:53"}, cfg.Recursors)
+		require.Equal(t, structs.RecursorStrategy("sequential"), cfg.RecursorStrategy)
+		require.False(t, cfg.AllowStale)
+		require.Equal(t, 20*time.Second, cfg.MaxStale)
+		require.Equal(t, 10*time.Second, cfg.NodeTTL)
+		ttl, _ := cfg.GetTTLForService("my_services_1")
+		require.Equal(t, 5*time.Second, ttl)
+		ttl, _ = cfg.GetTTLForService("my_specific_service")
+		require.Equal(t, 30*time.Second, ttl)
+		require.False(t, cfg.EnableTruncate)
+		require.False(t, cfg.OnlyPassing)
+		require.Equal(t, 15*time.Second, cfg.RecursorTimeout)
+		require.False(t, cfg.DisableCompression)
+		require.Equal(t, 1, cfg.ARecordLimit)
+		require.False(t, cfg.NodeMetaTXT)
+		require.Equal(t, uint32(1), cfg.SOAConfig.Refresh)
+		require.Equal(t, uint32(2), cfg.SOAConfig.Retry)
+		require.Equal(t, uint32(3), cfg.SOAConfig.Expire)
+		require.Equal(t, uint32(4), cfg.SOAConfig.Minttl)
```

Comment:
These assertions are specific to the V1 DNS server configuration. I made this a standalone V1 test and added a comment to make a version for V2 with different assertions.